src/plugins/dashboard_embeddable_container/public/embeddable/viewport/dashboard_viewport.tsx

```diff
@@ -82,24 +98,34 @@ export class DashboardViewport extends React.Component<DashboardViewportProps, S
     });
   };
 
+  private renderEmptyScreen() {
+    const emptyScreenProps = this.props.container.getInput().emptyScreenProps;
+    if (!emptyScreenProps) {
+      return null;
+    }
+    return <DashboardEmptyScreen {...emptyScreenProps} />;
+  }
+
   public render() {
     const { container } = this.props;
+    const { isFullScreenMode, isEmptyState, panels, title, description, useMargins } = this.state;
     return (
-      <div
-        data-shared-items-count={Object.values(this.state.panels).length}
-        data-shared-items-container
-        data-title={this.state.title}
-        data-description={this.state.description}
-        className={
-          this.state.useMargins ? 'dshDashboardViewport-withMargins' : 'dshDashboardViewport'
-        }
-      >
-        {this.state.isFullScreenMode && (
-          <this.context.services.ExitFullScreenButton
-            onExitFullScreenMode={this.onExitFullScreenMode}
-          />
-        )}
-        <DashboardGrid container={container} />
+      <div>
+        {isEmptyState ? this.renderEmptyScreen() : null}
```

Comment:
The empty screen needs to be inside dashboardViewport for it to render nicely, but mustn't be inside inside `data-shared-items-container` or else its content will be included in the report generation. This logic effectively means we could have emptyScreen AND dashboard panels rendered on the screen at the same time, but if the state is always reset appropriately, this should never happen.

Reply:
Instead of leaving it as a possibility to render both at the same time, should we specifically case the renderings? I wouldn't bet on state always being reset appropriately 😬 (though maybe I'm a pessimist).